src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs

```diff
@@ -420,6 +442,21 @@ public long GetMemtableSize()
 
         try
         {
+            if (_readAheadReadOptions != null && (flags & ReadFlags.HintReadAhead) != 0)
+            {
+                if (!readaheadIterators.IsValueCreated)
+                {
+                    readaheadIterators.Value = _db.NewIterator(cf, _readAheadReadOptions);
+                }
+
+                Iterator iterator = readaheadIterators.Value!;
+                iterator.Seek(key);
+                if (iterator.Valid() && Bytes.AreEqual(iterator.GetKeySpan(), key))
+                {
+                    return iterator.Value();
+                }
```

Comment:
Does it make sense to try doing everything in `Get` method? Instead using iterators explicitly where it make sense?

Reply:
I guess we can, but switching interface from trienoderesolver all the way to IDB is a bit much. Sadly, the readahead option only take into effect with an iterator.

Reply:
yes would have to pass some kind of access strategy through a lot of hoops. Still ThreadLocal looks very hacky - is it optimal even?

Reply:
Can't find other way easily. `ConcurrentBag` works also, but I'm guessing internally, it uses threadlocal also, might as well use it directly.

Reply:
Is it just for object pooling? `ConcurrentQueue` would give better reuse; `ThreadLocal` and `ConcurrentBag` will leave object on threads that might have been single use

Reply:
Ok, you should inherit from `ThreadLocal<Iterator>` then override `void Dispose (bool disposing)` so you can call `iterator.Dispose()` when its called by the finalizer because the thread has been exited

Reply:
Checked `ThreadLocal` source code. Not sure how that'll work. But I've wrapped the iterator into a managed class that call dispose on finalized. That should have the same effect.

Reply:
Something like that? https://github.com/NethermindEth/nethermind/pull/5645/commits/e81e0ba9efcbc51905a6898ffcd4c35180fbe256